src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -271,10 +270,14 @@ private Action<string> ErrorHandler
             {
                 return msg =>
                     {
-                        if (_projectContext != null)
+                        // instead of initializing it in MEF constructor since it access ISourceControlManagerProvider
+                        // initializing it here so that it doesn't hang at MEF initialization.
+                        if (ProjectContext == null)
```

Comment:
I thought about that as well, but realize it's only being used here so it will simple change to move it right here. Since it's a private API and has been called via Pumping JTF (since all install APIs invoke through pumping JTF) so it doesn't make any difference for thread safety. Do you think there could still be a missing edge case?